File: src/main/java/com/devspace/conexfy/converters/DevJsonToMapReadingConverter.java
package com.devspace.conexfy.converters;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

import org.springframework.core.convert.converter.Converter;
import org.springframework.data.convert.ReadingConverter;
import org.springframework.lang.NonNull;

import com.fasterxml.jackson.databind.ObjectMapper;

@ReadingConverter
public class DevJsonToMapReadingConverter implements Converter<String, Map<String, String>> {
    private static final ObjectMapper mapper = new ObjectMapper();

    @Override
    public Map<String, String> convert(@NonNull String source) {
        if (source.isBlank())
            return new HashMap<>();
        try {
            return mapper.readValue(source,
                    mapper.getTypeFactory().constructMapType(Map.class, String.class, String.class));
        } catch (IOException e) {
            throw new IllegalArgumentException(e);
        }
    }
}

File: src/main/kotlin/com/devspace/conexfy/converters/DevJsonToMapReadingConverter.kt
package com.devspace.conexfy.converters

import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.core.convert.converter.Converter
import org.springframework.data.convert.ReadingConverter
import org.springframework.lang.NonNull
import java.io.IOException

@ReadingConverter
class DevJsonToMapReadingConverter : Converter<String, MutableMap<String, String>> {
    override fun convert(@NonNull source: String): MutableMap<String, String> {
        if (source.isBlank()) return HashMap<String, String>()
        try {
            return mapper.readValue<MutableMap<String, String>>(
                source,
                mapper.typeFactory.constructMapType(MutableMap::class.java, String::class.java, String::class.java)
            )
        } catch (e: IOException) {
            throw IllegalArgumentException(e)
        }
    }

    companion object {
        private val mapper = ObjectMapper()
    }
}
